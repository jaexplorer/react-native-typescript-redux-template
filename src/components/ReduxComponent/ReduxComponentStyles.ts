import { StyleSheet } from 'react-native';
import { StylesCreator } from '../../lib/Theme';

const stylesCreator: StylesCreator = (theme, scale, moderateScale) =>
  StyleSheet.create({
    exampleContainer: {
      fontSize: moderateScale(14),
    },
  });

export default stylesCreator;
