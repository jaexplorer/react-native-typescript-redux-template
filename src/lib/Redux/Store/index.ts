import { createStore, combineReducers, applyMiddleware, Store } from 'redux';
import Thunk, { ThunkMiddleware } from 'redux-thunk';
import { persistStore, persistReducer, PersistConfig } from 'redux-persist';
import AsyncStorage from '@react-native-community/async-storage';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';

import { ExampleState, ExampleActionTypes } from '../Types/example';
import ExampleReducer from '../Reducers/example';
import { LoadingState, LoadingActionTypes } from '../Types/loading';
import LoadingReducer from '../Reducers/loading';

export interface ApplicationState {
  exampleState: ExampleState;
  loadingState: LoadingState;
}

const rootReducer = combineReducers<ApplicationState>({
  exampleState: ExampleReducer,
  loadingState: LoadingReducer,
});

export type AppActions = ExampleActionTypes | LoadingActionTypes;

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  stateReconciler: autoMergeLevel2,
  blacklist: ['loadingState'],
} as PersistConfig<ApplicationState>;

const persistedReducer = persistReducer(persistConfig, rootReducer);

const configureStore: Store<ApplicationState, AppActions> = createStore(
  persistedReducer,
  applyMiddleware(Thunk as ThunkMiddleware<ApplicationState, AppActions>),
);

export const persistor = persistStore(configureStore);

export default configureStore;
