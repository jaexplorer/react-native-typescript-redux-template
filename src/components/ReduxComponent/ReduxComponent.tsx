import React, { FC, useEffect } from 'react';
import { View } from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';
import { loadExample, LoadExampleResult } from '../../lib/Redux/Actions/example';
import { ApplicationState } from '../../lib/Redux/Store';
import { useTheme } from '../../lib/Theme';
import { Example } from '../../lib/Types';
import stylesCreator from './ReduxComponentStyles';

const ReduxComponent: FC<ReduxComponentProps> = ({ loading, examples, onLoadExample }) => {
  const [styles] = useTheme(stylesCreator);
  useEffect(() => {
    onLoadExample('123');
  }, []);

  return (
    <View style={styles.exampleContainer}>
      {loading ? (
        <View />
      ) : (
        <View>
          {examples.map((item) => (
            <View key={item.id} />
          ))}
        </View>
      )}
    </View>
  );
};

export interface ReduxComponentProps {
  loading: boolean;
  examples: Example[];
  onLoadExample: (exampleId: string | number) => LoadExampleResult;
}

const mapStateToProps = ({ loadingState, exampleState }: ApplicationState) => ({
  loading: loadingState.apiLoading,
  examples: exampleState.examples,
});

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      onLoadExample: loadExample as any,
    },
    dispatch,
  );

export default connect(mapStateToProps, mapDispatchToProps)(ReduxComponent);
