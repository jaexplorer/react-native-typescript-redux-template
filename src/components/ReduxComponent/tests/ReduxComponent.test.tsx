import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import ReduxComponent from '../ReduxComponent';
import Store from '../../../lib/Redux/Store';

test('Does component render test', () => {
  const { baseElement } = render(
    <Provider store={Store}>
      <ReduxComponent />
    </Provider>,
  );
  expect(baseElement).toMatchSnapshot();
});
