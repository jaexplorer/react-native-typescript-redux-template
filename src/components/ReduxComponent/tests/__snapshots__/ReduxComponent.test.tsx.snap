// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does component render test 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "fontSize": 22,
        }
      }
    >
      <View />
    </View>
  </View>
</View>
`;
